hs-abci-examples/nameservice/src/Nameservice/Modules/Nameservice/Keeper.hs

```diff
@@ -63,14 +63,17 @@ setName SetName{..} = do
         then throw $ UnauthorizedSet "Setter must be the owner of the Name."
         else do
           putWhois setNameName currentWhois {whoisValue = setNameValue}
-          BaseApp.emit NameRemapped
-             { nameRemappedName = setNameName
-             , nameRemappedNewValue = setNameValue
-             , nameRemappedOldValue = whoisValue
-             }
+          let event = NameRemapped
+                { nameRemappedName = setNameName
+                , nameRemappedNewValue = setNameValue
+                , nameRemappedOldValue = whoisValue
+                }
+          BaseApp.emit event
+          BaseApp.addContext (BaseApp.ContextEvent event) $
```

Comment:
this same expression appears multiple times, it's better to write some function like 

```haskell
logEvent
  :: Select e
  => Member Logging r
  => e
  -> Sem r ()
logEvent e =  BaseApp.addContext (BaseApp.ContextEvent e) $
  BaseApp.log BaseApp.Debug (cs $ BaseApp.makeEventType (Proxy :: Proxy e))
```

This probably lives in BaseApp.Events

Reply:
It's also probably an `Info` level log